Detour/Source/DetourNavMeshQuery.cpp

```diff
@@ -481,23 +482,24 @@ dtStatus dtNavMeshQuery::findRandomPointAroundCircle(dtPolyRef startRef, const f
 		v = &randomTile->verts[randomPoly->verts[j]*3];
 		dtVcopy(&verts[j*3],v);
 	}
-	
+
 	const float s = frand();
 	const float t = frand();
-	
+
 	float pt[3];
 	dtRandomPointInConvexPoly(verts, randomPoly->vertCount, areas, s, t, pt);
-	
+
 	float h = 0.0f;
 	dtStatus stat = getPolyHeight(randomPolyRef, pt, &h);
-	if (dtStatusFailed(status))
-		return stat;
-	pt[1] = h;
-	
+	if (dtStatusSucceed(stat)) {
+		// otherwise pt is on poly edge
+		pt[1] = h;
+	}
+
```

Comment:
I think this should be switched to `closestPointOnPoly` instead. Also, can you please undo the whitespace changes (or submit them as a separate fix)? Thanks!

Reply:
Got it